### ISSUE TITLE
Allow defining local indexes.

### ISSUE THREAD

**Issue by n1k0** [mozilla-services/kinto.js#66]
Filtering and ordering being currently achieved in-memory, the performance penalty can be importznt in case of large datasets.

We should allow users to create indexes on fields to speed things up.

Suggested API:

```js
const db = new Kinto();
const tasks = db.collection("tasks");
tasks.ensureIndexed("label", "done");
```

Thoughts?

**Comment by Natim** [comment]
I like this idea because it helps speed up things for schemaless.

**Comment by n1k0** [comment]
Something to pay attention to, because we'll alter the store schemas by adding/altering indexes, we'll need to handle database version numbers.

**Comment by n1k0** [comment]
Note: since #315 and #318 has now landed, we should implement custom index definition so users can benefit from the performance improvement provided by leveraging idb indexes for filtering and sorting.

**Comment by leplatrem** [comment]
https://twitter.com/daleharvey/status/735062319126982658

![capture d ecran de 2016-05-26 11-06-19](https://cloud.githubusercontent.com/assets/546692/15569610/ee157806-2331-11e6-9c22-a9eaa3084537.png)

**Comment by raevilman** [comment]
any plans for this..?


PS: Coming from DexieJS

**Comment by leplatrem** [comment]
Not really. But if someone contributes it, we would definitely provide support to ship it :)
